pkg/kubectl/cmd/get.go

```diff
@@ -244,7 +257,12 @@ func RunGet(f *cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []string
 			return err
 		}
 
-		if err := printer.PrintObj(obj, out); err != nil {
+                err = printer.PrintObj(obj, out)
+                if err == nil && len(args) >= 1 && strings.Contains(args[0], "pods") && cmd.Flag("show-all").Value.String() != "true" && cmd.Flag("no-headers").Value.String() != "true" {
```

Comment:
Pod specific checks do not belong in a generic command like RunGet.  It really needs to be abstracted into the printers, if anything.